hedera-node/src/main/java/com/hedera/services/queries/token/GetAccountNftInfoAnswer.java

```diff
@@ -0,0 +1,139 @@
+package com.hedera.services.queries.token;
+
+import com.hedera.services.context.primitives.StateView;
+import com.hedera.services.queries.AnswerService;
+import com.hedera.services.txns.validation.OptionValidator;
+import com.hedera.services.utils.SignedTxnAccessor;
+import com.hederahashgraph.api.proto.java.HederaFunctionality;
+import com.hederahashgraph.api.proto.java.Query;
+import com.hederahashgraph.api.proto.java.Response;
+import com.hederahashgraph.api.proto.java.ResponseCodeEnum;
+import com.hederahashgraph.api.proto.java.TokenGetAccountNftInfoQuery;
+import com.hederahashgraph.api.proto.java.TokenGetAccountNftInfoResponse;
+import com.hederahashgraph.api.proto.java.TokenNftInfo;
+
+import java.util.Map;
+import java.util.Optional;
+
+import static com.hedera.services.utils.SignedTxnAccessor.uncheckedFrom;
+import static com.hederahashgraph.api.proto.java.ResponseCodeEnum.INVALID_ACCOUNT_ID;
+import static com.hederahashgraph.api.proto.java.ResponseCodeEnum.INVALID_NFT_ID;
+import static com.hederahashgraph.api.proto.java.ResponseCodeEnum.OK;
+import static com.hederahashgraph.api.proto.java.ResponseType.COST_ANSWER;
+
+public class GetAccountNftInfoAnswer implements AnswerService {
```

Comment:
Keep in mind -> this might be refactored after https://github.com/hashgraph/hedera-services/pull/1616 as stated in: https://github.com/hashgraph/hedera-services/pull/1615#discussion_r653186368

Do not add unit tests for this class until we figure out the refactor changes